woodwork/serializers/serializer_base.py

```diff
@@ -0,0 +1,203 @@
+import datetime
+import json
+import os
+import tarfile
+import tempfile
+
+from woodwork.accessor_utils import _is_dask_dataframe, _is_koalas_dataframe
+from woodwork.logical_types import LatLong
+from woodwork.s3_utils import get_transport_params, use_smartopen
+from woodwork.type_sys.utils import _get_ltype_class, _get_specified_ltype_params
+from woodwork.utils import _is_s3, _is_url
+
+SCHEMA_VERSION = "11.3.0"
+
+PYARROW_IMPORT_ERROR_MESSAGE = (
+    f"The pyarrow library is required to serialize to {format}.\n"
+    "Install via pip:\n"
+    "    pip install pyarrow\n"
+    "Install via conda:\n"
+    "   conda install pyarrow -c conda-forge"
+)
+
+
+class Serializer:
+    def __init__(self, path, filename, data_subdirectory, typing_info_filename):
+        self.path = path
+        self.write_path = None
+        self.filename = filename
+        self.data_subdirectory = data_subdirectory
+        self.typing_info_filename = typing_info_filename
+        self.dataframe = None
+        self.typing_info = None
+        self.location = None
+        self.kwargs = {}
+
+    def serialize(self, dataframe, profile_name, **kwargs):
+        """Serialize data and typing information to disk."""
+        self.dataframe = dataframe
+        self.profile_name = profile_name
```

Comment:
It seems like `profile_name` only applies inside `save_to_s3`. Is there a reason we need to keep this outside of `save_to_s3`?

Reply:
Was not needed. Updated in [7cbba99](https://github.com/alteryx/woodwork/pull/1325/commits/7cbba99525ca9f0701caa6e2fe5d025c9fb0ed2b)

Reply:
let's also do the same for deserializing `read_from_s3`

Reply:
Good call. Done [6f5a6cc](https://github.com/alteryx/woodwork/pull/1325/commits/6f5a6ccef2f97b719e800e18141cc2ade51a88aa)